### PR TITLE
Add autoscale-v2 challenge to NYC Taxis track

### PR DESCRIPTION
**Pull request by gbanasiak**
This adds a new version of `autoscale` challenge where each step is defined with an array, instead of a fixed formula. The new version of the challange allows to specify both clients and target throughput, or only clients.

The challenge relies on bulk `looped` mode introduced in https://github.com/elastic/rally/pull/1830.

### CODE REVIEW

File: nyc_taxis/README.md [hunk 1]
```diff
@@ -78,6 +78,15 @@ This track allows to overwrite the following parameters using `--track-params`:
 * `include_force_merge` (default: true for non-serverless clusters, false for serverless clusters): Whether to include force merge operation.
 * `include_target_throughput` (default: true for non-serverless clusters, false for serverless clusters): Whether to apply target throughput.
 
+### Parameters specific to autoscale-v2 challenge
+
+This challenge is intended for serverless index tier autoscaler testing. It consists of configurable number of steps provided with array parameters.
+Note: `include_target_throughput` parameter is ignored in this challenge.
+
+* `as_clients` (default: []): An array with the number of indexing clients to be used in each step.
+* `as_time_periods` (default: []): An array with time period of every step. Time period is divided in halves. The first half is configured as warm-up.
```
Comment (inqueue) [inline nyc_taxis/README.md:87]:
```suggestion
* `as_time_periods` (default: []): An array with time period, in seconds, of every step. Time period is divided in halves. The first half is configured as warm-up.
```

File: nyc_taxis/challenges/common/autoscale-v2-schedule.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  "operation": {
+    "operation-type": "cluster-health",
+    "index": "nyc_taxis",
+    "request-params": {
+      "wait_for_status": "{{cluster_health | default('green')}}"
+    },
+    "retry-until-success": true
+  }
+}
+{%- set p_as_clients = (as_clients | default([]))%}
```
Comment (inqueue) [inline nyc_taxis/challenges/common/autoscale-v2-schedule.json:30]: What do you think about including some default values for these parameters? Something basic like `[8, 16, 32]` for clients and whatever makes sense for the other lists would be fine.

**Review by inqueue — APPROVED** [review]
I left a couple of comments. In all, LGTM.

File: nyc_taxis/challenges/common/autoscale-v2-schedule.json (hunk 2 above)
Comment (gbanasiak) [reply nyc_taxis/challenges/common/autoscale-v2-schedule.json]: Sure, addressed in https://github.com/elastic/rally-tracks/pull/569/commits/3c3d709f619a1b85556f82634134bcb8f2ca0205.

**Pull request merged**